composer/trainer/trainer.py

```diff
@@ -361,17 +397,37 @@ def create_from_hparams(cls, hparams: TrainerHparams) -> Trainer:
         train_dataloader = hparams.train_dataset.initialize_object(train_device_batch_size, hparams.dataloader)
 
         eval_device_batch_size = hparams.eval_batch_size // dist.get_world_size()
-        if hparams.val_dataset.shuffle and hparams.eval_subset_num_batches:
-            warnings.warn(
-                textwrap.dedent(f"""SubsetNumBatchesWarning: When specifying eval_subset_num_batches,
-            (set to {hparams.eval_subset_num_batches}), val_dataset.shuffle should be set to False. Otherwise,
-            each evaluation epoch may load a different subset of samples."""))
-        eval_dataloader = hparams.val_dataset.initialize_object(eval_device_batch_size, hparams.dataloader)
+        if hparams.val_dataset is not None:
```

Comment:
Probably don't need this since there are Evaluators now?

Reply:
I had thought maybe we'd leave the option for users to pass in a eval_dataloader and just create an evaluator from it in the trainer. That way, a user wouldn't have to interact with specifying metrics/evaluators at all if they have a simple model with just one evaluation dataset.